File: 20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    
  const brackets ={
   '(':')',
   '{':'}', 
   '[':']'   
  }
  
  const stack = []
  
  for (char of s){
    if(brackets[char]) stack.push(char)
    else if(stack.length > 0 && brackets[stack[stack.length - 1]] === char) stack.pop()
    else return false
  }
  
  return stack.length === 0
  
};

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        brackets = {
            '(':')',
            '{':'}',
            '[':']'
        }
        
        stack = []
        
        for char in s:
            if char in brackets:
                stack.append(char)
            elif (len(stack) > 0 and brackets[stack[len(stack) - 1]] == char):
                stack.pop()
            else:
                return False
        return len(stack) == 0